frontend/src/concepts/pipelines/content/compareRuns/CompareRunsRunList.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { ExpandableSection } from '@patternfly/react-core';
+import { TableVariant } from '@patternfly/react-table';
+import { PipelineRunKFv2 } from '~/concepts/pipelines/kfTypes';
+import { PipelinesFilter } from '~/concepts/pipelines/types';
+import { Table, useCheckboxTable } from '~/components/table';
+import CompareRunTableRow from '~/concepts/pipelines/content/compareRuns/CompareRunTableRow';
+import { compareRunColumns } from '~/concepts/pipelines/content/compareRuns/columns';
+import DashboardEmptyTableView from '~/concepts/dashboard/DashboardEmptyTableView';
+import usePipelineFilter, {
+  FilterOptions,
+  getDataValue,
+} from '~/concepts/pipelines/content/tables/usePipelineFilter';
+import CompareRunTableToolbar from '~/concepts/pipelines/content/compareRuns/CompareRunTableToolbar';
+
+type CompareRunsRunListProps = {
+  runs: PipelineRunKFv2[];
+  loading: boolean;
+};
+
+const CompareRunsRunList: React.FC<CompareRunsRunListProps> = ({ runs, loading }) => {
+  const [isExpanded, setExpanded] = React.useState(true);
+  const [, setFilter] = React.useState<PipelinesFilter | undefined>();
+  const filterToolbarProps = usePipelineFilter(setFilter);
+  const {
+    tableProps: checkboxTableProps,
+    toggleSelection,
+    isSelected,
+  } = useCheckboxTable(runs.map(({ run_id: runId }) => runId));
+
+  const filteredRuns = React.useMemo(
+    () =>
+      runs.filter((run) => {
+        const { filterData: data } = filterToolbarProps;
+        const runName = getDataValue(data[FilterOptions.NAME]);
+        const startedDateTime = getDataValue(data[FilterOptions.CREATED_AT]);
+        const state = getDataValue(data[FilterOptions.STATUS]);
+        const experimentId = getDataValue(data[FilterOptions.EXPERIMENT]);
+        const pipelineVersionId = getDataValue(data[FilterOptions.PIPELINE_VERSION]);
+        if (runName && !run.display_name.toLowerCase().includes(runName.toLowerCase())) {
+          return false;
+        }
+        if (startedDateTime && new Date(run.created_at) < new Date(startedDateTime)) {
+          return false;
+        }
+        if (state && run.state.toLowerCase() !== state.toLowerCase()) {
+          return false;
+        }
+        if (experimentId && run.experiment_id !== experimentId) {
+          return false;
+        }
+        if (
+          pipelineVersionId &&
+          run.pipeline_version_reference.pipeline_version_id !== pipelineVersionId
+        ) {
+          return false;
+        }
+        return true;
+      }),
+    [runs, filterToolbarProps],
+  );
+
+  return (
+    <ExpandableSection
+      toggleText="Run list"
+      isExpanded={isExpanded}
+      onToggle={(_, expanded) => setExpanded(expanded)}
+    >
+      {/* TODO: add max height to the table to show 5 rows at maximum */}
```

Comment:
this is a weird one to do, we may need to talk to UX to see if this is needed, or just have pagination start at 5? hard to calculate what the height is for 5 runs that dont have a fixed height

Reply:
for now I say we skip it and circle back